File: RegionalCostOfLiving.m
%% Regional Cost of Living Analysis
clear; clc; close all;

%% 1. Setup and Load
inFile = 'Regional Cost of Living.xlsx'; 
outDir = 'output'; 
if ~exist(outDir, 'dir'), mkdir(outDir); end

try
    T = readtable(inFile);
    disp('✅ File loaded successfully.');
catch ME
    error('Could not read file. Make sure "%s" is in your MATLAB folder.', inFile);
end

% Clean: Remove invalid years
if ismember('Year', T.Properties.VariableNames)
    T = T(T.Year > 1900, :); 
end

%% 2. Aggregate Data (FIXED)
% We list ONLY the numeric columns we want to average. 
% This prevents MATLAB from trying to average "Country" or "Region".
varsToAverage = {'Average_Monthly_Income', 'Cost_of_Living', ...
                 'Housing_Cost_Percentage', 'Tax_Rate', ...
                 'Healthcare_Cost_Percentage', 'Education_Cost_Percentage', ...
                 'Transportation_Cost_Percentage'};

% Calculate mean only for the variables listed above, grouped by Year
T_avg = groupsummary(T, 'Year', 'mean', varsToAverage);

disp('Data aggregated by Year.');

%% 3. Visualization 1: Income vs Cost
% Note: groupsummary adds 'mean_' to the column names
years = T_avg.Year;
income = T_avg.mean_Average_Monthly_Income;
cost   = T_avg.mean_Cost_of_Living;

fig1 = figure('Name', 'Income vs Cost', 'Position', [100 100 1000 600]);
set(fig1, 'ToolBar', 'none'); 

plot(years, income, '-o', 'LineWidth', 2, 'DisplayName', 'Avg Monthly Income');
hold on;
plot(years, cost, '-x', 'LineWidth', 2, 'DisplayName', 'Cost of Living');
hold off;

grid on;
title('Average Trends: Income vs. Cost of Living');
xlabel('Year');
ylabel('Amount (USD)');
legend('Location', 'best');

saveas(fig1, fullfile(outDir, 'Income_vs_Cost_Trend.png'));

%% 4. Visualization 2: Percentage Breakdown Stacked
percentage_cols = [T_avg.mean_Housing_Cost_Percentage, T_avg.mean_Tax_Rate, ...
                   T_avg.mean_Healthcare_Cost_Percentage, T_avg.mean_Education_Cost_Percentage, ...
                   T_avg.mean_Transportation_Cost_Percentage];

legend_names = {'Housing', 'Tax', 'Healthcare', 'Education', 'Transportation'};

fig2 = figure('Name', 'Cost Breakdown', 'Position', [100 100 1000 600]);
set(fig2, 'ToolBar', 'none');

bar(years, percentage_cols, 'stacked');

grid on;
title('Average Cost of Living Breakdown by Year');
xlabel('Year');
ylabel('Percentage of Income (%)');
legend(legend_names, 'Location', 'eastoutside');

saveas(fig2, fullfile(outDir, 'Cost_Breakdown_Stacked.png'));

disp('✅ Analysis complete. Graphs saved in "output" folder.');

File: python_files/Regional_Cost_of_Living.py
import pandas as pd
import matplotlib.pyplot as plt
from pathlib import Path
import os

def main():
    """Regional Cost of Living Analysis — Python version of MATLAB script"""

    # ---------------------------------------------------
    # 1. Setup and Load
    # ---------------------------------------------------
    in_file = "Regional Cost of Living.xlsx"
    out_dir = "output"

    Path(out_dir).mkdir(exist_ok=True)

    if not os.path.exists(in_file):
        raise FileNotFoundError(
            f'Could not read file. Make sure "{in_file}" is in the working directory.'
        )

    try:
        df = pd.read_excel(in_file)
        print("✅ File loaded successfully.")
    except Exception as e:
        raise RuntimeError(
            f'Could not read file. Ensure "{in_file}" exists and is a valid Excel file.'
        ) from e

    # ---------------------------------------------------
    # Clean: Remove invalid years
    # ---------------------------------------------------
    if "Year" in df.columns:
        df = df[df["Year"] > 1900]

    # ---------------------------------------------------
    # 2. Aggregate Data (Mean by Year)
    # ---------------------------------------------------
    vars_to_average = [
        "Average_Monthly_Income",
        "Cost_of_Living",
        "Housing_Cost_Percentage",
        "Tax_Rate",
        "Healthcare_Cost_Percentage",
        "Education_Cost_Percentage",
        "Transportation_Cost_Percentage"
    ]

    # Group by year and compute mean
    df_avg = df.groupby("Year")[vars_to_average].mean().reset_index()

    print("Data aggregated by Year.")

    # ---------------------------------------------------
    # 3. Visualization 1: Income vs Cost
    # ---------------------------------------------------
    years = df_avg["Year"]
    income = df_avg["Average_Monthly_Income"]
    cost = df_avg["Cost_of_Living"]

    plt.figure(figsize=(12, 6))

    plt.plot(years, income, "-o", linewidth=2, label="Avg Monthly Income")
    plt.plot(years, cost, "-x", linewidth=2, label="Cost of Living")

    plt.grid(True)
    plt.title("Average Trends: Income vs. Cost of Living")
    plt.xlabel("Year")
    plt.ylabel("Amount (USD)")
    plt.legend(loc="best")

    out_path1 = os.path.join(out_dir, "Income_vs_Cost_Trend.png")
    plt.savefig(out_path1, dpi=300, bbox_inches="tight")
    plt.close()

    # ---------------------------------------------------
    # 4. Visualization 2: Stacked Percentage Breakdown
    # ---------------------------------------------------
    percentage_cols = df_avg[
        [
            "Housing_Cost_Percentage",
            "Tax_Rate",
            "Healthcare_Cost_Percentage",
            "Education_Cost_Percentage",
            "Transportation_Cost_Percentage"
        ]
    ]

    legend_names = [
        "Housing",
        "Tax",
        "Healthcare",
        "Education",
        "Transportation"
    ]

    plt.figure(figsize=(12, 6))

    plt.bar(years, percentage_cols.values, stacked=True)

    plt.grid(True)
    plt.title("Average Cost of Living Breakdown by Year")
    plt.xlabel("Year")
    plt.ylabel("Percentage of Income (%)")
    plt.legend(legend_names, loc="center left", bbox_to_anchor=(1, 0.5))

    out_path2 = os.path.join(out_dir, "Cost_Breakdown_Stacked.png")
    plt.savefig(out_path2, dpi=300, bbox_inches="tight")
    plt.close()

    print("✅ Analysis complete. Graphs saved in 'output' folder.")
